### PR TITLE
Changes to work with the latest sw-testing-helpers

### PR DESCRIPTION
**Pull request by gauntface**
cc @wibblymat @surma @addyosmani - Medium sized PR for review please.

This updates Propel to using the latest sw-testing-helpers.

- Fixes TestServer binding to localhost rather than responding to external requests
- Access to all useful classes without requiring from node_modules
- Re-usable release and doc publishing scripts
- Doc template is shared and sorts revisioned directories correctly.

### CODE REVIEW

**Comment by surma** [comment]
LGTM :)

File: gulp-tasks/test.js [hunk 1]
```diff
@@ -18,12 +18,15 @@
 'use strict';
 
 const gulp = require('gulp');
-const spawn = require('child_process').spawn;
+var path = require('path');
```
Comment (wibblymat) [inline gulp-tasks/test.js]: `const`

**Comment by wibblymat** [comment]
LGTM

**Comment by addyosmani** [comment]
LGTM3

**Pull request merged**